src/follower.py

```diff
@@ -117,32 +128,47 @@ def reset_camera(self):
     will return False.
     """
     def detect(self):
-        obj_data = self._tag.detect()
-        detected = False
-
-        if obj_data:
-            detected = True
-            self._distance = obj_data['z']
-            self._turn_angle = np.degrees(obj_data['direction'])
-            self._decision = obj_data['decision']
+        tag_data = self._tag.detect()
 
-        return detected
+        if tag_data:
+            self._distance = tag_data['z']
+            self._turn_angle = np.degrees(tag_data['direction'])
+            self._decision = tag_data['decision']
+            return tag_data
+        else:
+            return False
 
+    """
+    Keeps the AR tag in view of the camera to avoid losing tag.
+    """
 
+    def track_tag(self):
+        tag_data = self.detect()
+        pan_angle = 5
+        
+        if tag_data == False:
+            self.stop()
+        elif tag_data['x'] < 0:
+            self._turn_camera_left(pan_angle)
+            self.turn()
+        elif tag_data['x'] > 0:
+            self._turn_camera_right(pan_angle)
+            self.turn()
+        
     """
     Follow an ARTag if one is found.
 
     If an ARTag is detected, the follower vehicle will turn towards it and
     manage speed to avoid collisions. If an ARTag is not detected, the vehicle
     will stop.
     """
+
     def follow(self):
         if self.detect():
             self.drive()
             self.turn()
         else:
-            self.stop()
-
+            self.track_tag()
```

Comment:
Change this back to self.stop() and move your code into turn().

Reply:
reverted all changes